src/SourceBuild/content/binary-tooling.sh

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env bash
```

Comment:
IMO having this separate from prep.sh hurts the usability.  prep is intended to prepare the repo/environment for a source-build.  Stripping binaries fits that charter.  For example binary stripping should be done by default when prep is run.  We shouldn't be requiring a separate script invocation.  That being said, prep as it stands today has possibly exceeded the capabilities of the current options design and may need a refactor.  That is outside the scope of this work which raises questions of how to add this functionality now while separating a possible refactor.

Reply:
> We shouldn't be requiring a separate script invocation.

Given that this script is primarily needed for distro maintainers, I don't see this as being a usability issue. They won't be using the prep script anyway (outside the initial bootstrap).

Reply:
The usability aspect is more for SB devs.  We should all be running the same configuration as distro maintainers.  Having a secondary optional script is going to lead devs to not build in this configuration.

More so, I have concerns about introducing an additional script as it doesn't scale.  We have seen continual growth in the prep like functionality.  Having separate scripts for each is not ideal.

Reply:
The reason I suggested that this be moved out of the prep script into its own script was precisely because of UX issues that it caused in the prep script. All of the parameters that this new script has will need to be stuffed into the prep script. So now we're forcing poor UX for distro maintainers (they'll need to provide a bunch of options to opt out of all the other stuff the prep script does by default).

Reply:
Yes, I agree that is a problem.  That is what I was eluding to when I said prep has been expanded enough where we have exceeded the original option design.  I think this addition warrants a proper redesign.  The question for me is how can we achieve this.  Do we do the re-design now as part of this PR, or is there a way to add it now and then do the redesign as a follow-up?  I would like to prioritize the single branch work.

Reply:
Ok, I'm fine with including it in the prep script if we make a plan for addressing the UX there.

Reply:
Opened an issue here: https://github.com/dotnet/source-build/issues/4181. I have combined the binary tooling scripting into the `prep-source-build.sh` script for now.